examples/hello_rust_cargo/CMakeLists.txt

```diff
@@ -0,0 +1,63 @@
+# ##############################################################################
+# apps/examples/hello_rust_cargo/CMakeLists.txt
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more contributor
+# license agreements.  See the NOTICE file distributed with this work for
+# additional information regarding copyright ownership.  The ASF licenses this
+# file to you under the Apache License, Version 2.0 (the "License"); you may not
+# use this file except in compliance with the License.  You may obtain a copy of
+# the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+# License for the specific language governing permissions and limitations under
+# the License.
+#
+# ##############################################################################
+
+include(nuttx_add_library)
+
+if(CONFIG_EXAMPLES_HELLO_RUST_CARGO)
+
+  # Call cargo build from CMakeLists and add it to the build system. Notice we
+  # should call cargo build with add_custom_target, otherwise cargo will be
+  # called every time when cmake is configured.
+
+  add_custom_command(
+    OUTPUT
+      ${CMAKE_BINARY_DIR}/hello_rust_cargo/riscv32imac-unknown-nuttx-elf/release/libhello.a
+    COMMAND
+      cargo build --release -Zbuild-std=core --manifest-path
```

Comment:
should we move to examples/hello_rust_cargo/hello/

Reply:
or better examples/rust/cargo/hello? :-)